src/ray/pubsub/publisher.cc

```diff
@@ -81,20 +75,18 @@ bool SubscriptionIndex::EraseSubscriber(const SubscriberID &subscriber_id) {
       key_id_to_subscribers_.erase(subscribers_it);
     }
   }
-  subscribers_to_key_id_.erase(subscribing_message_it);
+  subscribers_to_key_id_.erase(subscribing_key_it);
   return true;
 }
 
 bool SubscriptionIndex::EraseEntry(const std::string &key_id,
```

Comment:
Not related to this PR, but I feel the doc string should be improved [here](https://github.com/ray-project/ray/blob/41f80f641cf7e3587fadaf6ea3e79c5258df760f/src/ray/pubsub/publisher.h#L58-L59)

For example, if key_id == "", what doat that mean?

Reply:
Added comment.